app/main.py

```diff
@@ -2,24 +1,72 @@
+from abc import ABC, abstractmethod
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: type, name: str) -> None:
+        self.private_name = f"_{name}"
+
+    def __get__(self, instance: object, owner: type) -> None:
+        return getattr(instance, self.private_name, None)
+
+    def __set__(self, instance: object, value: object) -> None:
+        if not isinstance(value, int):
+            raise TypeError("Value must be an integer")
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError(f"Value must be between "
+                             f"{self.min_amount} and "
+                             f"{self.max_amount}")
+        setattr(instance, self.private_name, value)
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    @abstractmethod
+    def __init__(self, age: int, weight: int, height: int) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    def __init__(self, age: int, height: int, weight: int) -> None:
+        super().__init__(age, height, weight)
+
+    age = IntegerRange(4, 14)
+    height = IntegerRange(80, 120)
+    weight = IntegerRange(20, 50)
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    def __init__(self, age: int, height: int, weight: int) -> None:
```

Comment:
The parameter order in the `AdultSlideLimitationValidator` constructor does not match the parent class `SlideLimitationValidator`. Ensure the order is `age`, `weight`, `height` to maintain consistency.